esx_service/utils/auth.py

```diff
@@ -424,8 +398,17 @@ def authorize(vm_uuid, datastore_url, cmd, opts):
         - tenant_name: If the VM belongs to a tenant, return tenant_name, otherwise, return
         None
     """
-    logging.debug("Authorize: cmd=%s opts=`%s' vm_uuid=%s, datastore_url=%s", cmd, opts, vm_uuid, datastore_url)
-    # The possible value of "datastore_url" can be url of real datastore or "_VM_DS://"
+    # "datastore_url" is the url of datastore which the volume to be created on
+    # The possible value of "datastore_url" can be url of a real datastore or "_VM_DS"
+    # "privilege_ds_url" is the url of datastore which we need to get the privilege of
```

Comment:
I'm a bit confused here. Why datastore_url and privilege_ds_url can be different? In other words, why do we need to check the privilege of a different datastore other than the one on which the volume will be created?